plugins/processors/awsapplicationsignals/internal/resolver/attributesresolver.go

```diff
@@ -0,0 +1,162 @@
+	Stop(ctx context.Context) error
+}
+
+type attributesResolver struct {
+	subResolvers []subResolver
+}
+
+// create a new attributes resolver
+func NewAttributesResolver(resolvers []appsignalsconfig.Resolver, logger *zap.Logger) *attributesResolver {
+	subResolvers := []subResolver{}
+	for _, resolver := range resolvers {
+		switch resolver.Platform {
+		case appsignalsconfig.PlatformEKS, appsignalsconfig.PlatformK8s:
+			subResolvers = append(subResolvers, getKubernetesResolver(resolver.Platform, resolver.Name, logger), newKubernetesResourceAttributesResolver(resolver.Platform, resolver.Name))
+		case appsignalsconfig.PlatformEC2:
+			subResolvers = append(subResolvers, newResourceAttributesResolver(resolver.Platform, AttributePlatformEC2, DefaultInheritedAttributes))
+		default:
+			if ecsutil.GetECSUtilSingleton().IsECS() {
+				subResolvers = append(subResolvers, newResourceAttributesResolver(appsignalsconfig.PlatformECS, AttributePlatformGeneric, DefaultInheritedAttributes))
+			} else {
+				subResolvers = append(subResolvers, newResourceAttributesResolver(resolver.Platform, AttributePlatformGeneric, GenericInheritedAttributes))
+			}
+		}
+	}
+	return &attributesResolver{
+		subResolvers: subResolvers,
+	}
+}
+
+// Process the attributes
+func (r *attributesResolver) Process(attributes, resourceAttributes pcommon.Map, _ bool) error {
+	for _, subResolver := range r.subResolvers {
+		if err := subResolver.Process(attributes, resourceAttributes); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *attributesResolver) Stop(ctx context.Context) error {
+	var errs error
+	for _, subResolver := range r.subResolvers {
+		if err := subResolver.Stop(ctx); err != nil {
+			errs = errors.Join(errs, err)
+		}
+	}
+	return errs
+}
+
+type resourceAttributesResolver struct {
+	defaultEnvPrefix string
+	platformType     string
+	attributeMap     map[string]string
+}
+
+func newResourceAttributesResolver(defaultEnvPrefix, platformType string, attributeMap map[string]string) *resourceAttributesResolver {
+	return &resourceAttributesResolver{
+		defaultEnvPrefix: defaultEnvPrefix,
+		platformType:     platformType,
+		attributeMap:     attributeMap,
+	}
+}
+func (h *resourceAttributesResolver) Process(attributes, resourceAttributes pcommon.Map) error {
+	for attrKey, mappingKey := range h.attributeMap {
+		if val, ok := resourceAttributes.Get(attrKey); ok {
+			attributes.PutStr(mappingKey, val.Str())
+		}
+	}
+	if _, ok := attributes.Get(attr.AWSLocalEnvironment); !ok {
+		if val, found := resourceAttributes.Get(attr.AWSHostedInEnvironment); found {
+			attributes.PutStr(attr.AWSLocalEnvironment, val.Str())
+		} else {
+			if h.defaultEnvPrefix == appsignalsconfig.PlatformECS {
+				if clusterName, ok := getECSClusterName(resourceAttributes); ok {
+					attributes.PutStr(attr.AWSLocalEnvironment, GetDefaultEnvironment(h.defaultEnvPrefix, clusterName))
+				}
+			}
+			if h.defaultEnvPrefix == appsignalsconfig.PlatformEC2 {
```

Comment:
nit: Can be an else or switch-case. `h.defaultEnvPrefix` can't be equal to both ECS and EC2.

Reply:
Updated to `if else`